Sources_v3/StreamChat/Config/Token.swift

```diff
@@ -1,40 +1,61 @@
 //
-// Copyright © 2020 Stream.io Inc. All rights reserved.
+// Copyright © 2021 Stream.io Inc. All rights reserved.
 //
 
 import Foundation
 
-/// A token is used to authenticate a user.
-public typealias Token = String
+/// The type is designed to store the JWT and the user it is related to.
+public struct Token: Decodable, Equatable, ExpressibleByStringLiteral {
+    public let rawValue: String
+    public let userId: UserId
 
-/// A token provider is a function in which you send a request to your own backend to get a Stream Chat API token.
-///
-/// Set your custom `TokenProvider` in `ChatClientConfig` when creating a `ChatClient` instance. `ChatClient` will use it whenever
-/// it needs to get a new token for the given user.
-///
-public typealias TokenProvider = (_ apiKey: APIKey, _ userId: UserId, _ completion: @escaping (Token?) -> Void) -> Void
+    /// Created a new `Token` instance.
+    /// - Parameter value: The JWT string value. It must be in valid format and contain `user_id` in payload.
+    public init(stringLiteral value: StringLiteralType) {
+        try! self.init(rawValue: value)
```

Comment:
Used `fatalError` instead of our `log.assertion` since that's the `init` and it must have `self` with set `rawValue/userId` at the time it returns. But returning a token, in this case, doesn't make sense and we can/should crash (since it's used when the token is hardcoded as a string inside the app so we just notify the developer the token is invalid) 

e55f970b97ce750da86b4692041b7a03c1ee30b7